packages/datetime/test/dateRangeInputTests.tsx

```diff
@@ -122,12 +137,247 @@ describe("<DateRangeInput>", () => {
+                />);
+                root = result.root;
+
+                // clear the fields *before* setting up an onChange callback to
+                // keep onChange.callCount at 0 before tests run
+                changeStartInputText(root, "");
+                changeEndInputText(root, "");
+                root.setProps({ onChange });
+
+                return { root, onError };
+            });
+
+            afterEach(() => {
+                onChange = null;
+                onError = null;
+                root = null;
+            });
+
+            describe("shows the error message on blur", () => {
+                const _runTest = (input: WrappedComponentInput, inputString: string) => {
+                    changeInputText(input, inputString);
+                    input.simulate("blur");
+                    assertInputTextEquals(input, OUT_OF_RANGE_MESSAGE);
+                };
+                _runTestForEachScenario(_runTest);
+            });
+
+            describe("shows the offending date in the field on focus", () => {
+                const _runTest = (input: WrappedComponentInput, inputString: string) => {
+                    changeInputText(input, inputString);
+                    input.simulate("blur");
+                    input.simulate("focus");
+                    assertInputTextEquals(input, inputString);
+                };
+                _runTestForEachScenario(_runTest);
+            });
+
+            describe("calls onError with invalid date on blur", () => {
+                const _runTest = (input: WrappedComponentInput, inputString: string, boundary: DateRangeBoundary) => {
+                    const expectedRange = (boundary === DateRangeBoundary.START)
+                        ? [inputString, null]
+                        : [null, inputString];
+                    input.simulate("focus");
+                    changeInputText(input, inputString);
+                    expect(onError.called).to.be.false;
+                    input.simulate("blur");
+                    expect(onError.calledOnce).to.be.true;
+                    assertDateRangesEqual(onError.getCall(0).args[0], expectedRange);
+                };
+                _runTestForEachScenario(_runTest);
+            });
+
+            describe("does NOT call onChange before OR after blur", () => {
+                const _runTest = (input: WrappedComponentInput, inputString: string) => {
+                    input.simulate("focus");
+                    changeInputText(input, inputString);
+                    expect(onChange.called).to.be.false;
+                    input.simulate("blur");
+                    expect(onChange.called).to.be.false;
+                };
+                _runTestForEachScenario(_runTest);
+            });
+
+            describe("removes error message if input is changed to an in-range date again", () => {
+                const _runTest = (input: WrappedComponentInput, inputString: string) => {
```

Comment:
also no need for _ here; this is not a pattern we need in TS. 
and i'd inline these functions so you don't have to redeclare the types for each arg.

```js
runTestForEachScenario((input, inputString) => { ... })
```